pymeasure/instruments/agilent/agilentL4534A.py

```diff
@@ -0,0 +1,425 @@
+        range = Instrument.control(
+            "CONF:CHAN:RANG? (@{ch})",
+            "CONF:CHAN:RANG (@{ch}),%g",
+            """
+            Control Voltage range for this channel (0.25, 0.5, 1, 2, 4, 8, 16, 32, 128, 256).
+            """,
+            set_process=lambda v: assume_or_convert_units(v, ureg.V),
+            # send the value as V to the device
+            get_process=lambda v: ureg.Quantity(v, ureg.V),  # convert to quantity
+            validator=lambda value, values:
+                strict_discrete_set(assume_units(value, ureg.V), values),
+            values=VOLTAGE_RANGE_VALUES
+        )
+
+        coupling = Instrument.control(
+            "CONF:CHAN:COUP? (@{ch})",
+            "CONF:CHAN:COUP (@{ch}),%s",
+            """Comtrol channel coupling (AC|DC).""",
+            validator=strict_discrete_set,
+            values=COUPLING_VALUES
+        )
+
+        filter = Instrument.control(
+            "CONF:CHAN:FILT? (@{ch})",
+            "CONF:CHAN:FILT (@{ch}),%s",
+            """Control Filter for this channel (LP_20_MHZ, LP_2_MHZ, LP_200_KHZ).""",
+            validator=strict_discrete_set,
+            values=FILTER_VALUES
+        )
+
+        def _read_data_block(self, dtype):
+            """
+            Read data block header response that some FETCh commands return by default
+            :param dtype: Numpy data type expected for this data block
+            :return: numpy array of specified type
+            """
+            # Data block has header of form '#<n><n digits>',
+            # where <n>specifies the number of digits used to indicate the size of the block,
+            # and the <n digits> indicate the block size in bytes.
+            digits = self.read_bytes(2).decode()
+            if digits.startswith('#'):
+                try:
+                    bytes = int(self.read_bytes(int(digits[1])).decode())
+                    # Read specified number of bytes + newline terminator
+                    buf = memoryview(self.read_bytes(bytes+1))
+                    # Return view that strips off newline to contain only desired bytes
+                    return np.frombuffer(buf[:-1], dtype)
+                except ValueError:
+                    InstrumentError(
+                        """
+                        Unable to parse data block header returned by instrument.
+                        """
+                    )
+            else:
+                raise InstrumentError(
+                    """
+                    Data block header character "#" not found in response.
+                    Is device in ASCII mode?
+                    """
+                    )
+
+        @property
+        def voltage(self):
+            """Get voltage measurements for this channel (array of float in V)."""
+            self.write(f'FETC:WAV:VOLT? (@{self.id})')
+            return ureg.Quantity(self._read_data_block('>f4').astype(np.float32), ureg.V)
+
+        @property
+        def adc(self):
```

Comment:
I'm not happy with "adc", as it is not clear (from the name), what you intend.
Maybe "voltage_raw" (as it is the voltage measured, but returned in raw values).